deepspeed/utils/tp_shard.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+from deepspeed import comm as dist
+global num_kv_heads
```

Comment:
Would adding a global variable here be a good practice? Maybe you can make it a class attribute by adding a util class like [Loading](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/module_inject/auto_tp.py#L115)? Or combine these functions with [AutoTP](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/module_inject/auto_tp.py#L170) or [ReplaceWithTensorSlicing](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/module_inject/auto_tp.py#L19) ?

Reply:
get_shard_size and get_shard_size_list are called from three different files.  Wrap num_kv_heads in a class attribute will need a place to hold the instance.  Any suggstion how to ref to this class instance?

Reply:
I agree with @dc3671 that it would be nice to have these functions in [AutoTP](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/module_inject/auto_tp.py#L170) but that could create circular dependency with fusedqkv_utils.py. I think the module_inject folder is more appropriate than the utils folder though. I think adding these functions to auto_tp_model_utils.py would also be okay.

Reply:
Thanks @molly-smith I'll try move this file into module_inject folder.

Reply:
@molly-smith tp_shard.py had been moved into module_inject folder.